### PR TITLE
Templates doco

### PR DESCRIPTION
**Pull request by emiaj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,9 +38,15 @@ The only way to create a new Configuration is via Console (see [here](./docs/Con
 
 But the Document Generation Client does provide an API for editing an existing Configuration, and for running the document generation methods relative to that configuration and its descendent template storages.
 
+The response is a list of `DocumentGenerationTemplate` objects.  Each template includes a TemplateId string property and a Languages property which lists the relevant document languages as ISO 2 Letter Language code values.  E.g. en, da.
+
```
Comment (danielhazel) [inline README.md]: @emiaj 
Weird side effect of merge maybe ... I think this text is in another file now.

File: docs/Templates/Templates.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+Locate the `.docx` template file in your SharePoint instance and tag them accordingly (e.g. add the tag `invoice` as per our example above).  
+
+
+#### Configuration template subjects in Azure Blob Storage  
+
+The Azure Blob Storage provider from Logic Document Generation relies on the document metadata to filter out templates by `subject`.  
+
+You can set metadata for blobs using one of the following methods:
+
```
Comment (danielhazel) [inline docs/Templates/Templates.md]: @emiaj 
I think after reading this, I wouldn't be sure how to do this yet.
Azure Blob metadata is dictionary like.  key value pairs.
I think our code says that if a subject is included then we match blobs whose metadata contains a key equal to the subject, irrespective of the value given to that key.
I wonder if we should spell that out here.

File: docs/Templates/Templates.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+```csharp
+var templates = await documentGenerationClient.GetTemplates(subscriptionId, configurationId, hierarchyPath, "invoice").ConfigureAwait(false);
+```
+
+#### Configuring template subjects in SharePoint
+
+The SharePoint storage provider for Logic Document Generation relies on the document tags to filter out templates by `subject`.   
+
+Locate the `.docx` template file in your SharePoint instance and tag them accordingly (e.g. add the tag `invoice` as per our example above).
```
Comment (danielhazel) [inline docs/Templates/Templates.md]: @emiaj 
This surprises me.  But I haven't been inside sharepoint and seen what tags are.
The code seems to say that if a subject is included then we match files that are in a virtual subfolder with a name that matches the subject.
But maybe that's what tags are?  I'm ignorant.

File: docs/Templates/Templates.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+- Make sure you upload the file to the declared container and directory (`Blob prefix`) in Console for your configuration entry.  
+
+
+### How to hide a template from the list of available templates for a given entry
+
+#### Using SharePoint
+
+- Create an empty `.docx` file in your SharePoint Group.
+- Name it as the document you want to hide. 
+- For instance, if you want to hide `Template.docx`, the name of the document in your local SharePoint Group should be `Template.hidden.docx`
```
Comment (danielhazel) [inline docs/Templates/Templates.md]: @emiaj 
I might be wrong, I haven't tested the hidden thing, but I think the code says they should name it:

Template.docx.hidden

**Review by danielhazel — COMMENTED** [review]
With opinionated grammar-dar switched off.

File: docs/Templates/Templates.md (hunk 4 above)
Comment (danielhazel) [reply docs/Templates/Templates.md]: If the code does say that, and we want it to be like what you've written, we should just raise a bug.

Comment (emiaj) [reply docs/Templates/Templates.md]: Let me test this out first, I prefer to just adhere to whatever the convention is.

File: README.md (hunk 1 above)
Comment (emiaj) [reply README.md]: Fixed in 2e8b5e3e6424caff6ad8ea214a510d5a261c1995

File: docs/Templates/Templates.md (hunk 2 above)
Comment (emiaj) [reply docs/Templates/Templates.md]: > I wonder if we should spell that out here.
Yes, we use the metadata more like tags really.

Comment (emiaj) [reply docs/Templates/Templates.md]: Clarified in 598509c460e82b3490d43bafeb9c9cd498816c2c

File: docs/Templates/Templates.md (hunk 3 above)
Comment (emiaj) [reply docs/Templates/Templates.md]: Fixed misunderstanding in 98c27252849501b1792c0b69a4a5f2c77ca2bc5c

File: docs/Templates/Templates.md (hunk 4 above)
Comment (emiaj) [reply docs/Templates/Templates.md]: Fixed in 47b4a50856a1cc1a6875c717a25c1a6efe596a76

**Pull request merged**